test/index.js

```diff
@@ -234,6 +271,441 @@ describe('Schwifty', () => {
+                models: [
+                    TestModels.Dog,
+                    TestModels.Person
+                ]
+            }));
+
+            const plugin = {
+                name: 'my-plugin',
+                register: async (srv, opts) => {
+
+                    await srv.register({
+                        plugin: Schwifty,
+                        options: getOptions({
+                            migrationsDir: altMigrationsDir,
+                            models: []
+                        })
+                    });
+                }
+            };
+
+            await server.register(plugin);
+            await server.initialize();
+
+            let error;
+
+            try {
+                await server.plugins.schwifty.commands['migrate:diff'](server, ['test'], rootDir, internals.hpalCtx);
+            }
+            catch (err) {
+                error = err;
+            }
+
+            expect(error).to.exist();
+            expect(error.message).to.equal(`Only 1 migrations directory supported per project at this time. Please remove one of [${Path.join(__dirname, 'migrations/generated/root/migrations')},${Path.join(__dirname, 'migrations/generated2/root/migrations')}]`);
+
+            await server.stop();
+        });
+
+        it('throws asking the user to install schwifty-migrate-diff if not already installed', async (flags) => {
+
+            const originalGet = require('../lib/schwifty-migrate-diff').get;
+
+            require('../lib/schwifty-migrate-diff').get = () => null;
+
+            flags.onCleanup = () => {
+
+                require('../lib/schwifty-migrate-diff').get = originalGet;
+            };
+
+            const rootDir = Path.join(__dirname, 'migrations/generated/root');
+            const migrationsDir = rootDir + '/migrations';
+
+            const server = await getServer(getOptions({
+                migrationsDir,
+                models: [
+                    TestModels.Dog,
+                    TestModels.Person
+                ]
+            }));
+
+            let errHappened = false;
+
+            await server.initialize();
+            try {
+                await server.plugins.schwifty.commands['migrate:diff'](server, ['test'], rootDir, internals.hpalCtx);
+            }
+            catch (err) {
+                errHappened = true;
+                expect(err.message).to.equal('To use this command please first install `npm install --save-dev schwifty-migrate-diff`');
+            }
+
+            expect(errHappened).to.equal(true);
+            await server.stop();
+        });
+    });
+
+    describe('schwifty-migrate-diff require helper', () => {
+
+        it('returns null if schwifty-migrate-diff is not found', (flags) => {
+
+            expect(require('../lib/schwifty-migrate-diff').get('../bogus/path')).to.equal(null);
+        });
+
+        it('throws if trying to require schwifty-migrate-diff throws an error unrelated to requiring the module', (flags) => {
```

Comment:
I'm making these adjustments, but just so you know—if `get(true)` does _not_ throw then the test also passes.

Reply:
Bleh! I shoulda done the `let error;` dealio I did in others, set it in catch and ensure it's there after the try / catch block

Thanks!

Reply:
Try `expect(x).to.throw()`!